toolargetool/src/main/java/com/gu/toolargetool/ActivitySavedStateLogger.java

```diff
@@ -20,6 +22,7 @@
     @NonNull private final String tag;
     @Nullable private final FragmentSavedStateLogger fragmentLogger;
     @NonNull private final Map<Activity, Bundle> savedStates = new HashMap<>();
+    @NonNull private final List<FragmentActivity> createdActivities = new ArrayList<>();
```

Comment:
I think it would be best to avoid keeping references to activities like this if possible these references could cause memory leaks.

We already have a persistent reference to `fragmentLogger` so how about modifying the `FragmentSavedStateLogger` class to make it possible to disable logging through that? We could add a new method like `fragmentLogger.stopLogging()` which would set a flag inside the `FragmentSavedStateLogger` which would prevent it from actually logging anything.

Reply:
But we have already stored reference to activity through fragmentLogger callback.

Reply:
Yes, that is true, but the API provides some helpful guarantees for those references. The docs for `registerFragmentLifecycleCallbacks` say:

> All registered callbacks will be automatically unregistered when this FragmentManager is destroyed.

Besides the potential for leaking the references we could also avoid some book-keeping.

But anyway, this is all detail and I am happy to approve this either way so that `stopLogging` is added to the API. This code can always be refactored later if any problems become apparent.

Reply:
It's a good point about registerFragmentLifecycleCallbacks behaviour. I will change to flag. Thank you, Max.

Reply:
Cool, nice one Dmitry 👍

Reply:
I'm happy to trust your judgement because I you've been investigating this issue in more depth than me, which option do you think is best?

Personally, of those two options, I like the second one best and the reason is that I like the idea of the `stopLogging` API simply turning off the logging itself (i.e. calls to `Log.d`), without affecting how the lifecycle listeners are registered. If all the activity/fragment lifecycle listeners are always attached, whether logging is currently enabled or disabled ("stopped"), then it means that the logging can be turned off, or back on, at the flick of a switch. The logging can be turned on without having to wait for the components to go through any lifecycle events e.g. waiting for an activity to get to `onCreate` to re-register the fragment lifecycle listener.

Reply:
Second option looks better.